### PR TITLE
Add templating docs on logical operators

### PR DESCRIPTION
**Pull request by fwSmit**
Hi, i've added more templating documentation (and fixed a typo).

In #792 you mentioned a few issues surrounding template documentation. I've addressed the following ones:

- For #774, #787 and #790 I just pointed to the reference in the relevant places.
- #749 I fixed in my last PR
- #722 doesn't need to be documented
- I didn't do anything with #780 yet.

Are there any more  specific things you would like to see in the template documentation? 
I could add some complete examples at the end.

### CODE REVIEW

File: docs/TEMPLATING.md [hunk 1]
```diff
@@ -129,7 +224,7 @@ Create other files using the template
 `.local/share/chezmoi/small-font.yml.tmpl`
 
 ```
-{{- template "alacritty" 12 -}}
+{{- template "alacritty" 13 -}}
```
Comment (twpayne) [inline docs/TEMPLATING.md:227]: Was this change intended?

File: docs/TEMPLATING.md [hunk 2]
```diff
@@ -150,22 +245,33 @@ the `.` value passed in. You can test this with `chezmoi cat`:
...
+more arguments to the template, you can do it in two ways.
+
+#### Via chezmoi.toml
+
+This method is useful if you want to use the same template arguments multiple
+times, because you don't specify the arguments every time. Instead you specify
+them in the file `.chezmoi.toml`.
 
-`.config/chezmoi/chezmoi.toml`
+`.local/share/chezmoi/.chezmoi.toml`:
```
Comment (twpayne) [inline docs/TEMPLATING.md:258]: This should be `.config/local/chezmoi.toml`.

**Review by twpayne — COMMENTED** [review]
Thanks for this -- really appreciated. I think there a couple of nits.

File: docs/TEMPLATING.md (hunk 1 above)
Comment (fwSmit) [reply docs/TEMPLATING.md:227]: Nope, good spot

File: docs/TEMPLATING.md (hunk 2 above)
Comment (fwSmit) [reply docs/TEMPLATING.md:258]: Thanks, I've corrected it

**Comment by twpayne** [comment]
This is fantastic, thank you!
